File: backend/src/main/java/com/example/restaurant/entity/RestaurantTable.java
package com.example.restaurant.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "restaurant_table")
public class RestaurantTable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 50)
    private String name;

    @Column(nullable = false)
    private Integer capacity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private RestaurantZone zone;

    @Column(name = "x_position", nullable = false)
    private Integer xPosition;

    @Column(name = "y_position", nullable = false)
    private Integer yPosition;

    @Column(nullable = false)
    private Integer width;

    @Column(nullable = false)
    private Integer height;

    @Column(name = "window_side", nullable = false)
    private Boolean windowSide;

    @Column(name = "quiet_area", nullable = false)
    private Boolean quietArea;

    @Column(nullable = false)
    private Boolean accessible;

    @Column(name = "near_play_area", nullable = false)
    private Boolean nearPlayArea;
}

File: frontend/src/types/restaurant-table.ts
export type RestaurantZone = "MAIN" | "TERRACE" | "PRIVATE";

export type RestaurantTable = {
  id: number;
  name: string;
  capacity: number;
  zone: RestaurantZone;
  xPosition: number;
  yPosition: number;
  width: number;
  height: number;
  windowSide: boolean;
  quietArea: boolean;
  accessible: boolean;
  nearPlayArea: boolean;
};

export type RestaurantTableStatus = RestaurantTable & {
  occupied: boolean;
};
